components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -696,6 +700,34 @@ private String getCaptchaParams(String tenantDomain) {
                                 log.debug("Enforcing recaptcha for resend confirmation is not enabled.");
                             }
                         }
+                    } else if (captchaEnabledConfigName.equals(connectorConfig.getName())) {
+                        if (Boolean.parseBoolean(connectorConfig.getValue())) {
+                            Property[] maxFailedConfig = BasicAuthenticatorDataHolder.getInstance()
+                                    .getIdentityGovernanceService()
+                                    .getConfiguration(new String[]{maxFailedAttemptCaptchaConfigName}, tenantDomain);
+                            Property maxFailedProperty = maxFailedConfig[0];
+                            try {
+                                if (Integer.valueOf(maxFailedProperty.getValue()) < failedLoginAttempts) {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Number of failed attempts is higher than max failed " +
+                                                "attempts for reCaptcha. Recaptcha will be enforced");
+                                    }
+                                    captchaParams += BasicAuthenticatorConstants.RECAPTCHA_PARAM + "true";
+                                } else {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Number of failed attempts is less than max failed " +
+                                                "attempts for reCaptcha. Recaptcha will not be enforced");
+                                    }
+                                }
+                            } catch (NumberFormatException e) {
```

Comment:
Is it safe to continue the flow and allow the user to access if this configuration is incorrect ?

Reply:
yes there is an issue... This cause due to admin users error in configuring the value. So failing the login is not the best solution as I think. We can use the default value in such cases. Will fix this

Reply:
What's the value set for Max failed attempts if maxFailedProperty throws a NumberFormatException?

Reply:
3 is the default value

Reply:
If we are continuing with the default value if the user hasn't configured the value properly , let's mention that in the logs.

Reply:
+1

Reply:
fixed https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/121